### PR TITLE
Support 'empty' prop and type 'any'

### LINKED ISSUE #22: Add 'empty' global validator property
If true, empty strings are allowed. If false, field must contain something, or be undefined/null (if not required). Default to false.

### LINKED ISSUE #23: Add 'any' type
Allows an assert that a field exists in any form, without specifying what it should be.

### PR DESCRIPTION
**Pull request by Fluidbyte**
Closes #23 and Closes #22

### CODE REVIEW

**Comment by stephen-ta** [comment]
:accept:

File: src/types/array.js [hunk 1]
```diff
@@ -4,9 +4,15 @@ import ValidationError from '../lib/error'
...
 const array = {
   default: context => {
+    // Ensure array
     if (!Array.isArray(context.value)) {
       return context.fail('Value must be an array')
     }
+    // If empty (and empty allowed), move forward
+    if (context.def.empty && context.value.length === 0) {
+      return context.value
+    }
```
Comment (TomFrost) [inline src/types/array.js]: We're not failing in the event that empty isn't allowed, and context.value is an empty array

**Comment by TomFrost** [comment]
Strings and arrays were already allowed to be empty before this push -- it would be good to have a test around the failure case for each of these, to prove empty values can be blocked.

File: src/types/array.js (hunk 1 above)
Comment (Fluidbyte) [inline src/types/array.js]: @TomFrost: added with [7f361d1](https://github.com/TechnologyAdvice/obey/commit/7f361d12d52300c8b0b50c2c3bda7ada952233f9)

**Comment by Fluidbyte** [comment]
@TomFrost I'll write some integration tests to cover your last comment since it will need to move into the actual type checks.

**Pull request merged**